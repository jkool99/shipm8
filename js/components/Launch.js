import React from 'react';
import { View, StyleSheet, TouchableOpacity, Text, Image } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

// Load FontAwesome icons
// Icon.loadFont();

const Launch = ({ navigation }) => {
  const getStarted = () => {
    navigation.navigate('Login');
  };
  return (
    <View style={styles.container} behavior="padding" enabled>
      <Text style={styles.banner}>WELCOME</Text>

      <Image
        source={require('../../assets/shipm8_logo.png')}
        style={styles.logo}
      />
      <View style={styles.logoContainer}>
        <Text style={styles.textStyle}>
          Monitor Your K8s Clusters Anywhere!
        </Text>
        <View style={styles.formContainer} />
        <View style={{ paddingTop: 5 }}>
          <TouchableOpacity
            style={styles.buttonContainer}
            activeOpacity={0.7}
            behavior="padding"
            enabled
            onPress={getStarted}>
            <Text style={styles.buttonText}>Get Started</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
};

export default React.memo(Launch);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    justifyContent: 'center',
    alignItems: 'center',
    paddingBottom: 60,
  },
  buttonContainer: {
    backgroundColor: '#1589FF',
    paddingVertical: 15,
    borderRadius: 5,
    marginTop: 10,
  },
  banner: {
    fontSize: 50,
    marginBottom: 50,
    fontWeight: 'bold',
    color: '#151B54',
  },
  buttonText: {
    textAlign: 'center',
    color: 'white',
    fontWeight: '700',
    fontSize: 16,
  },
  textStyle: {
    textAlign: 'center',
    alignContent: 'center',
    justifyContent: 'center',
    alignItems: 'center',
    color: '#151B54',
    fontSize: 18,
    fontWeight: 'bold',
    marginTop: 5,
  },
  logo: {
    width: 315,
    height: 315,
    alignContent: 'center',
    alignItems: 'center',
    borderColor: '#151B54',
    borderWidth: 3,
    marginBottom: 25,
  },
  formContainer: {
    paddingVertical: 25,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
