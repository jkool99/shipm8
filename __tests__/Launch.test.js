import 'react-native';
import React from 'react';
import Launch from '../js/components/Launch';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Launch Component', () => {
  it('renders correctly', () => {
    const launchTest = renderer.create(<Launch />).toJSON();
    expect(launchTest).toMatchSnapshot();
  });
});
