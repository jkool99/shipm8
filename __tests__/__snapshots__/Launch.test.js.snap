// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch Component renders correctly 1`] = `
<View
  behavior="padding"
  enabled={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 60,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#151B54",
        "fontSize": 50,
        "fontWeight": "bold",
        "marginBottom": 50,
      }
    }
  >
    WELCOME
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/shipm8_logo.png",
      }
    }
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "borderColor": "#151B54",
        "borderWidth": 3,
        "height": 315,
        "marginBottom": 25,
        "width": 315,
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "color": "#151B54",
          "fontSize": 18,
          "fontWeight": "bold",
          "justifyContent": "center",
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      Monitor Your K8s Clusters Anywhere!
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingVertical": 25,
        }
      }
    />
    <View
      style={
        Object {
          "paddingTop": 5,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#1589FF",
            "borderRadius": 5,
            "marginTop": 10,
            "opacity": 1,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "700",
              "textAlign": "center",
            }
          }
        >
          Get Started
        </Text>
      </View>
    </View>
  </View>
</View>
`;
